sdk/eventhub/event-hubs/src/log.ts

```diff
@@ -20,3 +20,62 @@ export function logErrorStackTrace(error: unknown): void {
     logger.verbose(error.stack);
   }
 }
+
+/**
+ * Creates a logger that includes the connectionId, sender or receiver name, and
+ * client type.
+ * @internal
+ */
+export function createLogPrefix(
+  connectionId?: string,
+  type?: "Sender" | "Receiver" | "Management",
+  name?: string
+): string {
+  let prefix = "";
+  if (connectionId) {
+    prefix = `[${connectionId}]`;
+  }
+  if (type) {
+    prefix += ` ${type}`;
+  }
+  if (name) {
+    prefix += ` [${name}]`;
+  }
+  return prefix;
+}
+
+type LogFunction = Debugger["log"];
+
+/**
+ * @internal
+ */
+export interface SimpleLogger {
+  readonly info: LogFunction;
+  readonly error: LogFunction;
+  readonly warning: LogFunction;
+  readonly verbose: LogFunction;
+}
+
+function createLogFunction(
+  azureLogger: AzureLogger,
+  prefix: string,
+  level: AzureLogLevel
+): (arg: any, ...args: any[]) => void {
+  return (arg: any, ...args: any[]) =>
+    azureLogger[level](
+      ...(typeof arg === "string" ? [prefix + " " + arg] : [prefix, arg]),
```

Comment:
nit 
```suggestion
      ...(typeof arg === "string" ? [`${prefix} ${arg}`] : [prefix, arg]),
```

Reply:
Addressed in https://github.com/Azure/azure-sdk-for-js/pull/25214/commits/9e1faf1551c63799b5ad35689ebb081699067fa1.